ts/pages/governance/treasury.tsx

```diff
@@ -0,0 +1,395 @@
+import * as React from 'react';
+import ReactMarkdown from 'react-markdown';
+import { useQuery } from 'react-query';
+import { Redirect, useParams } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Banner } from 'ts/components/banner';
+import { Button } from 'ts/components/button';
+import { DocumentTitle } from 'ts/components/document_title';
+import { RegisterBanner } from 'ts/components/governance/register_banner';
+import { Column, Section } from 'ts/components/newLayout';
+import { StakingPageLayout } from 'ts/components/staking/layout/staking_page_layout';
+import { Heading, Paragraph } from 'ts/components/text';
+import { Text } from 'ts/components/ui/text';
+import { Countdown } from 'ts/pages/governance/countdown';
+import { TreasuryProposal } from 'ts/pages/governance/data';
+import { ModalTreasuryVote } from 'ts/pages/governance/modal_vote';
+import { VoteInfo, VoteValue } from 'ts/pages/governance/vote_form';
+import { VoteStats } from 'ts/pages/governance/vote_stats';
+import { colors } from 'ts/style/colors';
+import { WebsitePaths } from 'ts/types';
+import { configs, GOVERNANCE_THEGRAPH_ENDPOINT } from 'ts/utils/configs';
+import { documentConstants } from 'ts/utils/document_meta_constants';
+import { utils } from 'ts/utils/utils';
+
+const FETCH_PROPOSAL = gql`
+  query proposal($id: ID!) {
+      proposal(id: $id) {
+        id
+        proposer
+        description
+        votesFor
+        votesAgainst
+        createdTimestamp
+        voteEpoch {
+          id
+          startTimestamp
+          endTimestamp
+        }
+        executionEpoch {
+          id
+          startTimestamp
+          endTimestamp
+        }
+        executionTimestamp
+      }
+    }
+`;
+
+type ProposalState = 'created' |'active' |'succeeded' |'failed' |'queued' |'executed';
+
+export const Treasury: React.FC<{}> = () => {
+    const [proposal, setProposal] = React.useState<TreasuryProposal>();
+    const [isProposalsLoaded, setProposalsLoaded] = React.useState<boolean>(false);
+    const [isVoteReceived, setIsVoteReceived] = React.useState<boolean>(false);
+    const [isVoteModalOpen, setIsVoteModalOpen] = React.useState<boolean>(false);
+    const [quorumThreshold, setQuorumThreshold] = React.useState<BigNumber>();
+    const { id: proposalId } = useParams();
+
+    const { data } = useQuery('proposal', async () => {
+        const { proposal: proposalFromGraph } = await request(GOVERNANCE_THEGRAPH_ENDPOINT, FETCH_PROPOSAL, {
+            id: proposalId,
+        });
+
+        return proposalFromGraph;
+    });
+
+    React.useEffect(() => {
+        // tslint:disable-next-line:no-floating-promises
+        (async () => {
+            // const qThreshold = await contract.quorumThreshold().callAsync();
+            // setQuorumThreshold(qThreshold);
+            const hardCodeQuorumThreshold = new BigNumber('10000000');
+            setQuorumThreshold(hardCodeQuorumThreshold);
+        })();
```

Comment:
With the new smart contract integration this would be picked up from the contract. Have left this here as a reminder that this is where we can fetch this. Will add a TODO here for better documentation. Will that work?